aries-site/src/pages/_app.js

```diff
@@ -59,15 +60,159 @@ const backgroundImages = {
  * The `Component` prop is the active `page`, so whenever you
  * navigate between routes, `Component` will change to the new `page`.
  */
+
+export const ViewContext = createContext(undefined);
+
 function App({ Component, pageProps, router }) {
   const route = router.route.split('/');
 
-  // necessary to ensure SkipLinks can receive first tab focus
-  // after a route change
+  //state that holds the update information within the last 30 days
+  const [wholeViewHistory, setWholeViewHistory] = useState({});
+  //state that holds boolean for whether or not update info is ready to be rendered
+  const [pageUpdateReady, setPageUpdateReady] = useState(false);
+
+  //this effect is only for the first time a page loads
+  useEffect(() => {
+    let name = router.asPath;
+    let nameArray = name.split('/');
+    name = nameArray[Object.keys(nameArray).length - 1];
+    name = name.charAt(0).toUpperCase() + name.slice(1);
+    name = name.split('#')[0];
+    name = name.split('?')[0];
+
+    //imported and put fetchData in here so that it can access the setWholeViewHistory function
+    let thirtyDaysAgo = new Date().getTime() - 30 * 24 * 60 * 60 * 1000;
+    fetch(
+      `https://api.github.com/repos/grommet/hpe-design-system/pulls?state=closed`,
+    )
+      .then(response => response.json())
+      .then(data => {
+        let temporaryHistory = {};
+        let tokenName;
+        for (let i = 0; i < Object.keys(data).length; i++) {
+          if (new Date(data[i].merged_at).getTime() < thirtyDaysAgo) {
+            //if it is older than thirty days ago
+            break;
+          }
+          let temporaryString = data[i].body;
+          if (temporaryString.includes('#### Notifications')) {
+            const indexOfFirstComponent =
+              temporaryString.search('#### Notifications') + 22;
```

Comment:
Why 22?

Reply:
I also commented it in there because i realized it was vague, but it needed to be pushed 22 characters forward from the first index in order to get it to the component names and update types. For example, it's generally like "#### Notifications\r\n" that it needs to get through

Reply:
Would it be helpful to extra `#### Notifications` into a variable and then reference its length? That would also clean up how many times a repeated string is used.

For example:

```
const notificationsHeading = "#### Notifications";

...

if (temporaryString.includes(notificationsHeading)) {
   const indexOfFristComponent = temporaryString.search(notificationsHeading) + notificationsHeading.length
}
```